src/api/app/models/user.rb

```diff
@@ -696,16 +697,7 @@ def delete!
   end
 
   def involved_projects_ids
-    # just for maintainer for now.
-    role = Role.hashed['maintainer']
-
-    ### all projects where user is maintainer
-    projects = relationships.projects.where(role_id: role.id).pluck(:project_id)
-
-    # all projects where user is maintainer via a group
-    projects += Relationship.projects.where(role_id: role.id).joins(:groups_users).where(groups_users: { user_id: id }).pluck(:project_id)
-
-    projects.uniq
+    (relationships.projects.maintainers.pluck(:project_id) + group_relationships.projects.maintainers.pluck(:project_id)).uniq
```

Comment:
After some discussions: I still think this is more explicit and and on my machine it was even slightly faster (even that stackoverflow says that they almost equal)